Sources/OfflineEntitlements/PurchasedProductsFetcher.swift

```diff
@@ -43,3 +48,16 @@ class PurchasedProductsFetcher {
     }
 
 }
+
+// MARK: -
+
+/// An empty implementation of `PurchasedProductsFetcherType` for platforms
+/// that don't support `PurchasedProductsFetcher`.
+final class VoidPurchasedProductsFetcher: PurchasedProductsFetcherType {
+
+    func fetchPurchasedProducts() async throws -> [PurchasedSK2Product] {
+        assertionFailure("This should never be used")
```

Comment:
This will only crash in `DEBUG` builds. We only create this prior to iOS15, where offline `CustomerInfo`s aren't computed.